sample_apps/simple_app/run_example.sh

```diff
@@ -156,6 +156,26 @@ function run_cmd() {
    set +x
 }
 
+# ###########################################################################
+# Deep-clean the build-env to remove artifacts (e.g. generated files, binaries
+# etc.) that may have been produced by other steps. We run this to ensure
+# that this script will run successfully w/o any dependencies on executing
+# some prior steps.
+# ###########################################################################
+function rm_non_git_files() {
+    pushd "${CERT_PROTO}" > /dev/null 2>&1
+
+    echo "${Me}: Delete all files not tracked by git"
+    # shellcheck disable=SC2046
+    run_cmd rm -rf $(git ls-files . --exclude-standard --others)
+
+    echo "${Me}: Delete all files not tracked by git that are also ignored"
+    # shellcheck disable=SC2046
+    run_cmd rm -rf $(git ls-files . --exclude-standard --others --ignored)
```

Comment:
NOTE: I have manually tested these two `git ls-files` commands, and they _seem_ to work well-enough for our current usages.

I am no Git-master ,... so am relying on CI-clean test runs to validate that this change is acceptable.